lib/rouge/lexers/reasonml.rb

```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8 -*- #
+# frozen_string_literal: true
+
+module Rouge
+  module Lexers
+    load_lexer 'ocaml/common.rb'
+
+    class ReasonML < OCamlCommon
+      title "ReasonML"
+      desc 'New syntax on top of OCaml ecosystem (reasonml.github.io)'
+      tag 'reasonml'
+      filenames '*.re', '*.rei'
+      mimetypes 'text/x-reasonml'
+
+      def self.keywords
+        @keywords ||= super + Set.new(%w(
+          switch
+        ))
+      end
+
+      state :root do
+        rule %r/\s+/m, Text
+        rule %r/false|true|[(][)]|\[\]/, Name::Builtin::Pseudo
+        rule %r/#{@@upper_id}(?=\s*[.])/, Name::Namespace, :dotted
+        rule %r/`#{@@id}/, Name::Tag
```

Comment:
What does it mean to demarcate an identifier with `` ` `` in ReasonML?

Reply:
(Just trying to understand if `Name::Tag` is the most appropriate token.)

Reply:
It's a [polymorphic variant](https://v1.realworldocaml.org/v1/en/html/variants.html#polymorphic-variants) - more or less a tagged value. Makes sense to me.